### PR TITLE
VB-3651 when session template reference passed is null visits are returned for all prisons

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: docker-compose-local.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ services:
       - FEATURE_EVENTS_SNS_ENABLED=false
 
   visit-scheduler-db:
-    image: postgres:13.2
+    image: postgres:15.5
```
Comment (dhirajshettymoj) [inline docker-compose-local.yml:50]: non related change - corrected version for postgres

File: docker-compose-localstack.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version: "3"
 services:
 
   localstack:
-    image: localstack/localstack:1.3
+    image: localstack/localstack:3.0
```
Comment (dhirajshettymoj) [inline docker-compose-localstack.yaml:6]: non related change - corrected version for localstack

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt [hunk 3]
```diff
@@ -87,14 +87,23 @@ class VisitSchedulerClient(
       .bodyToMono<List<VisitDto>>().block(apiTimeout)
   }
 
-  fun getVisitsForSessionTemplateAndDate(sessionTemplateReference: String?, sessionDate: LocalDate, visitStatusList: List<VisitStatus>, visitRestrictions: List<VisitRestriction>?, page: Int, size: Int): RestPage<VisitDto>? {
+  fun getVisitsForSessionTemplateAndDate(
+    sessionTemplateReference: String?,
+    sessionDate: LocalDate,
+    visitStatusList: List<VisitStatus>,
+    visitRestrictions: List<VisitRestriction>?,
+    prisonCode: String,
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt:95]: add prison code as a parameter when getting visits by sessionTemplateReference

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt [hunk 4]
```diff
@@ -87,14 +87,23 @@ class VisitSchedulerClient(
...
+    size: Int,
+  ): RestPage<VisitDto>? {
     return webClient.get()
       .uri("/visits/session-template") {
         it.queryParamIfPresent("sessionTemplateReference", Optional.ofNullable(sessionTemplateReference))
           .queryParam("fromDate", sessionDate)
           .queryParam("toDate", sessionDate)
           .queryParamIfPresent("visitRestrictions", Optional.ofNullable(visitRestrictions))
           .queryParam("visitStatus", visitStatusList)
+          .queryParam("prisonCode", prisonCode)
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt:106]: adding prisonCode to visit-scheduler call

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationVisitsController.kt [hunk 5]
```diff
@@ -331,8 +331,12 @@ class OrchestrationVisitsController(
     @RequestParam
     @NullableNotEmpty
     visitRestrictions: List<VisitRestriction>?,
+    @Schema(name = "prisonCode", description = "Filter results by prison id/code", example = "MDI", required = true)
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/OrchestrationVisitsController.kt:334]: add prison code as a parameter when getting visits by sessionTemplateReference

**Review by tpmcgowan — APPROVED** [review]
LGTM

**Pull request merged**